DataConnectors/CEF/cef_gather_info.py

```diff
@@ -0,0 +1,166 @@
+advanced_command_dict = {
+    "top_processes": ["sudo top -bcn1 -w512", "head -n 20"],
+    "omsagent_process": ["sudo ps -aux", "grep omsagent"]
+}
+
+
+def print_notice(input_str):
+    '''
+    Print given text in white background
+    :param input_str:
+    '''
+    print("\033[0;30;47m" + input_str + "\033[0m")
+
+
+def append_content_to_file(command_object, file_path=output_file_path):
+    """
+
+    :param command_object: consists of the name and the output
+    :param file_path: a file to share the commands outputs
+    """
+    output = repr(command_object).replace('%', '%%')
+    command_tokens = ["sudo", "bash", "-c", "printf '" + "\n" + output + "' >> " + file_path]
+    try:
+        write_new_content = subprocess.Popen(command_tokens, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        write_new_content.communicate()
+    except Exception:
+        print(str(command_object.command) + "was not documented successfully")
+
+
+def run_command(command):
+    """
+
+    :param command: the key value pair from the command dict
+    :return: a command object consisting of a name and an output
+    """
+    command_to_run = subprocess.Popen(basic_command_dict[command].split(' '), stdout=subprocess.PIPE,
+                                      stderr=subprocess.STDOUT)
+    try:
+        o, e = command_to_run.communicate()
+    except Exception:
+        print(basic_command_dict[command] + "failed to run")
+    o = o.decode(encoding='UTF-8')
+    command_object = SystemInfo(command, o)
+    return command_object
+
+
+def run_special_command(command):
+    """
+
+    :param command: the key value pair from the special command dict
+    :return: a command object consisting of a name and an output
+    """
+    first_command = subprocess.Popen(advanced_command_dict[command][0].split(' '), stdout=subprocess.PIPE,
+                                     stderr=subprocess.STDOUT)
+    second_command = subprocess.Popen(advanced_command_dict[command][1].split(' '), stdin=first_command.stdout,
+                                      stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    try:
+        o, e = second_command.communicate()
+    except Exception:
+        print(command + "command failed to run")
+    o = o.decode(encoding='UTF-8')
+    command_object = SystemInfo(command, o)
+    return command_object
+
+
+def clean_up(path=output_file_path):
+    """
+    :param path: A path to delete
+    """
+    clean_up_command = subprocess.Popen(['rm', path, '2>', '/dev/null'],
+                                        stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    try:
+        o, e = clean_up_command.communicate()
+    except Exception:
+        print("Clean up command failed to run")
+
+
+def main():
+    print_notice("Note this script should be run in elevated privileges")
+    print("Beginning to collect server data- This may take a few minutes")
+    clean_up()
+    for command in basic_command_dict.keys():
```

Comment:
I'd suggest to use tqdm here, but this will require library installation, so maybe consider writing at the beginning how many commands there are?

Reply:
Will add in Future versions.
Currently runs instantly so no need for that.